Fix #25 - New game flow

mgoimam opened the pull request:
- Start with no game set and a welcome screen.
- New command to start a new game [Enter/Return on desktops, Start on
  gamepads].
- Add null check in UDPRenderer since game can now be null.
- Simplified the logic in the AggregateGameInput to make adding new
  commands a bit less cumbersome.

mgoimam committed:
Fix #25 - New game flow

- Start with no game set and a welcome screen.
- New command to start a new game [Enter/Return on desktops, Start on
  gamepads].
- Add null check in UDPRenderer since game can now be null.
- Simplified the logic in the AggregateGameInput to make adding new
  commands a bit less cumbersome.

scjody commented on AppletRenderer.pde:
```diff
@@ -26,14 +20,24 @@ class AppletRenderer implements ControlListener {
+      if (currentGame == null) {
+        text("READY TO PLAY", width/2, height/2 - 35);
+      } else if (currentGame.isGameOver()) {
+        text("GAME OVER\nSCORE: " + currentGame.getScore(), width/2, height/2 - 50);
```
Can you add the lines count here too please?

scjody commented on UDPRenderer.pde:
```diff
@@ -15,6 +15,8 @@ class UDPRenderer {
   
   void renderGameState(TetrisGame currentGame) {
+    if (currentGame == null) return;
+    
```
Can you actually send an emptyGrid() in this case too?

scjody commented:
Also can you send an emptyGrid() on game quit?

mgoimam committed:
Improve #25 - New game flow

- Display the number of lines as well as the score on the game over
  screen.
- Make the grid height and width constants.
- Send an empty grid via UDP while there is no game active (before the
  first game and just before quitting with ctrl+q).

mgoimam commented:
I'm only clearing the UDP renderer when the program is quit using the CTRL+Q shortcut. It should be possible to hook onto the application exit otherwise, but I'd need to research that a bit more. Also, I don't think doing it via the Processing app will ever be 100% reliable. If the application ever crashes or hangs, we still have a timeout on the server side that shuts off the valves, right?

scjody commented:
Yeah, CTRL-Q is what I meant.  You're right, it's not actually necessary, it just means we'll turn off the fire a tiny bit sooner...

scjody commented:
Great, :shipit: !
